pkg/gitversion/gitversion.go

```diff
@@ -66,10 +66,15 @@ func GetLanguageVersions(workingDirPath string, commitish plumbing.Revision) (*L
 		}
 	}
 
-	// Detect if the git worktree is dirty, and add `.dirty` to the version if it is
+	// Detect if the git worktree is dirty, and add `dirty` to the version if it is
 	if versionComponents.Dirty {
-		preVersion = fmt.Sprintf("%s.dirty", preVersion)
-		pythonPreVersion = fmt.Sprintf("%s+dirty", pythonPreVersion)
+		if versionComponents.IsExact {
+			preVersion = fmt.Sprintf("%s+dirty", preVersion)
+			pythonPreVersion = fmt.Sprintf("%s+dirty", pythonPreVersion)
+		} else {
+			preVersion = fmt.Sprintf("%s.dirty", preVersion)
+			pythonPreVersion = fmt.Sprintf("%s+dirty", pythonPreVersion)
```

Comment:
this needs to be `.dirty` AFAICT?

Reply:
This is actually correct, unfortunately :(
```
 go run cmd/pulumictl/main.go util get-version --language javascript                                                                
v0.0.2-alpha.1589842781+d14cbec3.dirty
 go run cmd/pulumictl/main.go util get-version --language python                                                                    
0.0.2a1589842781+dirty
```

It's because python's semver is weird